src/parquet/column/properties.h

```diff
@@ -151,77 +226,147 @@ class PARQUET_EXPORT WriterProperties {
 
-  int64_t dictionary_pagesize() const { return dictionary_pagesize_; }
+  inline int64_t dictionary_pagesize() const { return dictionary_pagesize_; }
 
-  int64_t data_pagesize() const { return pagesize_; }
+  inline int64_t data_pagesize() const { return pagesize_; }
 
-  ParquetVersion::type version() const { return parquet_version_; }
+  inline ParquetVersion::type version() const { return parquet_version_; }
 
-  Encoding::type encoding(const std::shared_ptr<schema::ColumnPath>& path) const {
-    Encoding::type coding = default_encoding_;
+  inline std::string created_by() const { return parquet_created_by_; }
+
+  inline const ColumnEncodingSet& encodings(
+      const std::shared_ptr<schema::ColumnPath>& path) const {
     auto it = encodings_.find(path->ToDotString());
-    if (it != encodings_.end()) { coding = it->second; }
-
-    // Use the correct enum value for dictionary coding based on the used Parquet version
-    if (coding == Encoding::PLAIN_DICTIONARY || coding == Encoding::RLE_DICTIONARY) {
-      if (parquet_version_ == ParquetVersion::PARQUET_1_0) {
-        return Encoding::PLAIN_DICTIONARY;
-      } else {
-        return Encoding::RLE_DICTIONARY;
-      }
-    }
-    return coding;
+    if (it != encodings_.end()) { return it->second; }
+    return default_encodings_;
+  }
+
+  inline Encoding::type rep_level_encoding(
+      const std::shared_ptr<schema::ColumnPath>& path) const {
+    auto it = encodings_.find(path->ToDotString());
+    if (it != encodings_.end()) { return it->second.rep_level_encoding; }
+    return default_rep_level_encoding_;
+  }
+
+  inline Encoding::type def_level_encoding(
+      const std::shared_ptr<schema::ColumnPath>& path) const {
+    auto it = encodings_.find(path->ToDotString());
+    if (it != encodings_.end()) { return it->second.def_level_encoding; }
+    return default_def_level_encoding_;
   }
 
-  Compression::type compression(const std::shared_ptr<schema::ColumnPath>& path) const {
+  inline Encoding::type dictionary_encoding(
+      const std::shared_ptr<schema::ColumnPath>& path) const {
+    auto it = encodings_.find(path->ToDotString());
+    if (it != encodings_.end()) { return it->second.dictionary_encoding; }
+    return default_dictionary_encoding_;
+  }
+
+  inline Encoding::type value_encoding(
+      const std::shared_ptr<schema::ColumnPath>& path) const {
+    auto it = encodings_.find(path->ToDotString());
+    if (it != encodings_.end()) { return it->second.value_encoding; }
+    return default_value_encoding_;
+  }
+
+  inline Encoding::type dictionary_fallback_encoding(
+      const std::shared_ptr<schema::ColumnPath>& path) const {
+    auto it = encodings_.find(path->ToDotString());
+    if (it != encodings_.end()) { return it->second.dictionary_fallback_encoding; }
+    return default_dictionary_fallback_encoding_;
+  }
+
+  inline Compression::type compression(
+      const std::shared_ptr<schema::ColumnPath>& path) const {
```

Comment:
Before you needed to call the method multiple times -- that seems OK to me. For example:

```
prop_builder->encoding(path1, enc1)
    ->encoding(path2, enc2)
    ->encoding(path3, enc3)
    ->encoding(path4, enc4)
    ->encoding(path5, enc5)
```